### ISSUE TITLE
[Bug]: No video acceleration on nightly using VAAPI on AMDGPU even after #4532

### ISSUE THREAD

**Issue by RodoMa92** [FreeTubeApp/FreeTube#4647]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

Even after adding that specific flag on Chromium, on AMDGPU it still doesn't seem to want to work. Vaapi confirms that the drivers are there, see

```
Trying display: wayland
libva info: VA-API version 1.20.0
libva info: Trying to open /usr/lib64/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_20
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.20 (libva 2.20.1)
vainfo: Driver version: Mesa Gallium driver 23.3.4 for AMD Custom GPU 0405 (radeonsi, vangogh, LLVM 17.0.6, DRM 3.54, 6.6.14-202.fsync.fc39.x86_64)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSlice
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSlice
      VAProfileHEVCMain               : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointEncSlice
      VAProfileHEVCMain10             : VAEntrypointVLD
      VAProfileHEVCMain10             : VAEntrypointEncSlice
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileVP9Profile0            : VAEntrypointVLD
      VAProfileVP9Profile2            : VAEntrypointVLD
      VAProfileAV1Profile0            : VAEntrypointVLD
      VAProfileNone                   : VAEntrypointVideoProc
```

But no hardware acceleration is active after playing a video, only CPU. Verified with the media tab from the developer menu.

### Expected Behavior

VAAPI should decode video on the GPU, and not the CPU.

### Issue Labels

inconsistent behavior

### FreeTube Version

v0.19.1-beta

### Operating System Version

Fedora Silverblue

### Installation Method

Flathub

### Primary API used

Invidious API

### Nightly Build

- [X] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Revival8697 thoughts?

**Comment by Revival8697** [comment]
I unfortunately do not own any AMD GPUs.

You can try experimenting with flags mentioned [here](https://wiki.archlinux.org/title/Chromium#Hardware_video_acceleration).

It may be an [upstream issue](https://issues.chromium.org/issues/40267998). If that were the case, there is a [workaround](https://bbs.archlinux.org/viewtopic.php?pid=2133444#p2133444).

**Comment by absidue** [comment]
According to your log you are using wayland, which means this issue is a duplicate of #4621.

**Comment by Revival8697** [comment]
> According to your log you are using wayland, which means this issue is a duplicate of #4621.

FreeTube runs on XWayland by default so that isn't an issue, but OP should clarify.

**Comment by RodoMa92** [comment]
Oh, didn't knew that there was still missing pieces in chromium for supporting wayland. I'll wait for Electron 29 to drop then.

VAAPI only support native X11 then?

**Comment by Revival8697** [comment]
> VAAPI only support native X11 then?

Yes. Did you run FreeTube in native Wayland?

**Comment by RodoMa92** [comment]
Nope, I haven't forced anything. I just tried to force x11 using `--ozone-platform-hint=x11` and it is still rendering on CPU. Interesting logged error in console that I just noticed:
```
[86399:0206/091020.557134:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[86399:0206/091021.983489:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[86399:0206/091027.444833:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
[86399:0206/091130.053364:ERROR:vaapi_video_decoder.cc(1226)] failed Initialize()ing the frame pool
```

**Comment by absidue** [comment]
I don't know much about Linux stuff but didn't Fedora say they were going to go wayland only? So xwayland wouldn't be available?

**Comment by absidue** [comment]
@RodoMa92 Can you try running the command mentioned here to check if it's actually running through xwayland? https://askubuntu.com/a/1393619

**Comment by RodoMa92** [comment]
Same issue forcing X11 on latest nightly, no video acceleration to be seen anywhere and same exact error on vaapi initialization as xwayland on wayland.

I'm not using Silverblue directly, but using a custom distribution on top of it (Bazzite Silverblue).

**Comment by absidue** [comment]
Unless you happen to find a magic set of flags that make it work, there is nothing else we can do on the FreeTube side, other than wait for upstream (chromium) to add more support or fix related bugs. That being said it's worth noting that the chromium developer's stance on VA-API on Linux is the following:
> VA-API on Linux is not supported, but it can be enabled using the flags below, and it might work on certain configurations -- but there's no guarantees.
https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/gpu/vaapi.md#vaapi-on-linux

> Insofar as Linux + VA-API is not a supported platform (IOW it's a best
effort) and this will allow developers and users to try and make it work
on their machines.
https://chromium-review.googlesource.com/c/chromium/src/+/3646633

**Comment by RodoMa92** [comment]
Yeah, `xlsclients` confirms that it's running on xwayland:
```
~  xlsclients 
DeckDiMarco  steamwebhelper
DeckDiMarco  freetube
DeckDiMarco  steam
```

**Comment by RodoMa92** [comment]
> Unless you happen to find a magic set of flags that make it work, there is nothing else we can do on the FreeTube side, other than wait for upstream (chromium) to add more support or fix related bugs. That being said it's worth noting that the chromium developer's stance on VA-API on Linux is the following:
> 
> > VA-API on Linux is not supported, but it can be enabled using the flags below, and it might work on certain configurations -- but there's no guarantees.
> > https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/gpu/vaapi.md#vaapi-on-linux
> 
> > Insofar as Linux + VA-API is not a supported platform (IOW it's a best
> > effort) and this will allow developers and users to try and make it work
> > on their machines.
> > https://chromium-review.googlesource.com/c/chromium/src/+/3646633

Yeah, there is no combination of flags that I can find that seems to enable them again on Freetube. I have no clue on why they can't implement something that just works at least for VAAPI without breaking every single version..

**Comment by RodoMa92** [comment]
I'll probably wait for Electron 29 to be merged in and test again with Chromium 122, but for now I can't find any magical flags to make it works. Thanks, Google.

**Comment by Revival8697** [comment]
> failed Initialize()ing the frame pool

That sounds like [the upstream bug](https://issues.chromium.org/issues/40267998) then. You can try [the workaround](https://bbs.archlinux.org/viewtopic.php?pid=2133444#p2133444) I mentioned.

**Comment by absidue** [comment]
> I have no clue on why they can't implement something that just works at least for VAAPI without breaking every single version..

Presumably because it's very much still work-in-progress and if you look at the quoted messages above, they make zero promises that it will work or keep working. So instead of being angry that it doesn't work, they literally told you it probably won't, you should assume by default that it won't work or will break at any moment and then be positively surprised if it does happen to work for you at some point.
